File: Programmers/콜라츠 추측.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;
int solution(int num) {
	long long a = num;
	int answer = 0;
	while (a != 1) {
		if (a % 2 == 0)
		{
			a /= 2;
		}
		else
		{
			a *= 3;
			a++;
		}
		answer++;
		if (answer == 500)
		{
			return -1;
		}
	}
	return answer;
}

int main() {
	int num;
	cin >> num;
	cout << solution(num);
	system("pause");
}

File: Programmers/콜라츠 추측.java
class Solution {
    public int solution(int num) {
        int tmp = 0;
        while(true) {
            if(num == 1)
                return tmp;
            else if(tmp == 500)
                return -1;
            else if(num % 2 == 0)
                num /= 2;
            else if(num % 2 == 1) {
                num *= 3;
                num += 1;
            }
            tmp++;
        }
    }
}